conans/client/manager.py

```diff
@@ -97,6 +98,17 @@ def deps_install(app, ref_or_path, install_folder, base_folder, graph_info, remo
     conanfile.folders.set_base_imports(install_folder)
     conanfile.folders.set_base_generators(base_folder)
 
+    if local_folder:
+        for cf in conanfile.dependencies.host.values():  # TODO: build requires? collisions suffix?
+            _name = cf.ref.name
+            if conanfile.settings.get_safe("build_type"):
```

Comment:
Probably the ``build_type`` not necessary here if we actually leverage the ``layout()``, as it can be seen in the test below with ``path = os.path.join(client.current_folder, "cmake-build-release", "conan")`` path.

 I didn't think of that, but maybe it fits nicely, just using the layout definition, taking advantage that is already paremeterized for different build_type (and possibly architectures, or even other custom configs)

On the other hand, having it separately from the build folders could be nice, easier to .gitignore the build, or clean the build more easily, without actually removing the deps.

Reply:
It is necessary for multi config, where the dependency binary packages must be different (that's why the suffix) but the generators folder according to the layout won't. (build/conan)

Reply:
Yes, it would be something equivalent to ``libdirs`` that it will have either the base folder conditioned (``cmake-build-release``), or the lib folder conditioned ``build/conan/Release``. Sounds something that could be parameterized and modeled in ``layout``